contracts/MyOXO.sol

```diff
@@ -0,0 +1,282 @@
+pragma solidity 0.4.15;
+
+contract MyOXO {
+
+// по аналогии с  https://github.com/Purka
+
+	address public dan;//игрок 1
+	address public kate;// игрок2
+
+	byte[10] public field;
+/* масcив игрового поля
+	[1][2][3]
+	[4][5][6]
+	[7][8][9]
+*/
+
+	byte public lastMove = "O";
+
+	mapping (address => bool) public payed;
+	mapping (address => uint) public wallets;
+
+	uint public step = 0;
+	uint public deposit = 1000 wei; // ставка каждого игрока
+
+	uint lastMoveTime;
+
+	enum StateMachine {GameEnded, Game} // ограничитель событий игры
+	StateMachine public state;
+
+
+	modifier onlyPlayer() {
+		if (msg.sender != dan && msg.sender != kate) {
+			return;
+		}
+		_;
+	}
+
+	modifier onlyDuringTheGame() {
+		if (state != StateMachine.Game) {
+			return;
+		}
+		_;
+	}
+
+
+	function getState() constant public returns (StateMachine){
+		return state;
+	}
+
+	function Balance() constant public returns (uint) {
+		return this.balance;
+	}
+
+	function getField(uint x) constant public returns (byte) {
+		return field[x];
+	}
+
+// события в игре
+
+	event Message(string message);
+	event Deposit(uint value);
+	event Win(string message, address winner);
+	event Draw(string message);
+	event Withdraw(address player, uint money);
+
+//Игроки делают ставки в размере 1000 wei
+
+	function deposit() payable public returns (bool success) {
+		require(state == StateMachine.GameEnded);
+		require(msg.value == deposit);
+		require(!payed[msg.sender]);
+
+//Кто первый заплатил, тот ходит первым
+
+		if (dan == 0) {
+			dan = msg.sender;
+			payed[msg.sender] = true;
+			Deposit(msg.value);
+			return true;
+		}
+		if (payed[dan] && kate == 0) {
+			kate = msg.sender;
+			payed[msg.sender] = true;
+			Deposit(msg.value);
+
+//Старт игры
+
+			state = StateMachine.Game;
+			lastMoveTime = now;
+			return true;
+		}
+		return false;
+	}
+
+// ходы делаются только игроками и только в игровое время
+	function move(uint x) onlyPlayer onlyDuringTheGame public returns (bool success) {
+
+//Блокировка двойного хода
+			if (withinTheField(x) && EmptyCell(x)) {

```

Comment:
You don't need to check withinTheField(). But you should make the field bytes[9] instead of bytes[10]. So in this case you will have 9 cells, from 0 to 8